domain/consensus/processes/consensusstatemanager/add_block_to_virtual.go

```diff
@@ -40,7 +27,36 @@ func (csm *consensusStateManager) AddBlockToVirtual(blockHash *externalapi.Domai
 	return nil
 }
 
+func (csm *consensusStateManager) resolveBlockStatusAndCheckFinality(blockHash *externalapi.DomainHash) error {
+	isNextVirtualSelectedParent, err := csm.isNextVirtualSelectedParent(blockHash)
+	if err != nil {
+		return err
+	}
+
+	if !isNextVirtualSelectedParent {
+		return nil
+	}
+
+	blockStatus, err := csm.resolveBlockStatus(blockHash)
+	if err != nil {
+		return err
+	}
+
+	if blockStatus == externalapi.StatusValid {
```

Comment:
Not critical if you disagree, but I'd rather:
1. rename this method to maybeResolveBlockStatus
2. move the `if statusValid { checkFinalityViolation}` out of it